scripts/generateAssetData/generateRelatedAssetIndex/generateRelatedAssetIndex.ts

```diff
@@ -229,11 +250,17 @@ export const generateRelatedAssetIndex = async () => {
   const generatedAssetData: AssetsById = JSON.parse(
     await fs.promises.readFile(generatedAssetsPath, 'utf8'),
   )
-  const relatedAssetIndex: Record<AssetId, AssetId[]> = {}
-  const assetDataWithRelatedAssetKeys: AssetsById = { ...generatedAssetData }
+  const relatedAssetIndex: Record<AssetId, AssetId[]> = JSON.parse(
+    await fs.promises.readFile(relatedAssetIndexPath, 'utf8'),
+  )
+  const assetDataWithRelatedAssetKeys: Record<AssetId, PartialFields<Asset, 'relatedAssetKey'>> = {
+    ...generatedAssetData,
+  }
 
-  // remove relatedAssetKey from the existing data to ensure the related assets get updated
-  Object.values(assetDataWithRelatedAssetKeys).forEach(asset => delete asset.relatedAssetKey)
+  if (rebuildAll) {
+    // remove relatedAssetKey from the existing data to ensure the related assets get updated
+    Object.values(assetDataWithRelatedAssetKeys).forEach(asset => delete asset.relatedAssetKey)
```

Comment:
Might be stupid but why if the `rebuildAll` prop is passed, we don't just nuke the file and recreate it from scratch?

Reply:
That's effectively what we do! Agree would be a bit more performant not to read the original file and assume empty one, but prefer not to add additional branching here 🙏🏽